doc/source/extending.rst

```diff
@@ -135,6 +135,14 @@ There are two approaches for providing operator support for your ExtensionArray:
 2. Use an operator implementation from pandas that depends on operators that are already defined
    on the underlying elements (scalars) of the ExtensionArray.
 
+.. note::
+
+   Regardless of the approach, you may want to implement ``__array_ufunc__``
+   or set ``__array_priority__`` if you want your implementation
+   to be called when involved in binary operations with NumPy
```

Comment:
Is ``__array_ufunc__`` directly used for binary operations?

If you use the ExtensionOpsMixin to set all the dunder methods, I don't think you need `__array_ufunc__` for that? As I understood it, you can implement all the operators with `__array_ufunc__` using the NDArrayOperatorsMixin (https://github.com/numpy/numpy/blob/v1.15.1/numpy/lib/mixins.py#L63-L183) (but of course it is still recommendable to implement it for ufuncs?)

Just to say that I find the note not fully clear.

Also, do you know if `__array_priority__` is still used for other things? (or is it only used for ufuncs? In which case it can be clearer that this is for older numpy versions?)

Reply:
> Also, do you know if __array_priority__ is still used for other things?

I don't know. I only mentioned `__array_ufunc__`, because it disables `__array_priority__`.

Reply:
> As I understood it, you can implement all the operators with __array_ufunc__ using the NDArrayOperatorsMixin 

You may be saying this, but `NDAraryOperatorsMixin` implements all the special methods by calling `__array_ufunc__`. So if your array implements `__array_ufunc__`, you can get all the operators for free.

But implementing `__array_ufunc__` doesn't necessitate using `NDArrayOperatorsMixin`.  I'm not sure what happens on a class defining `__add__`, `__array_ufunc__`, and `__array_priority__`,